apps/cowswap-frontend/src/common/updaters/orders/UnfillableOrdersUpdater/updaters/UnfillableOrderUpdater.ts

```diff
@@ -0,0 +1,100 @@
+import { useAtom, useSetAtom } from 'jotai'
+import { useRef } from 'react'
+
+import { useIsWindowVisible } from '@cowprotocol/common-hooks'
+import { isSellOrder } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+
+import useSWR, { SWRConfiguration } from 'swr'
+
+import { PENDING_ORDERS_PRICE_CHECK_POLL_INTERVAL } from 'legacy/state/orders/consts'
+
+import { updatePendingOrderPricesAtom } from 'modules/orders/state/pendingOrdersPricesAtom'
+
+import { useIsProviderNetworkUnsupported } from '../../../../hooks/useIsProviderNetworkUnsupported'
+import { useUpdateIsUnfillableFlag } from '../hooks/useUpdateIsUnfillableFlag'
+import { fetchOrderPrice } from '../services/fetchOrderPrice'
+import { orderLastTimePriceUpdateAtom } from '../state/orderLastTimePriceUpdateAtom'
+import { OrderToCheckFillability } from '../types'
+
+const SWR_CONFIG: SWRConfiguration = {
+  // Since we use orderLastTimePriceUpdateAtom to avoid frequent quote updates
+  // We can try to refresh twice frequent, because it won't do a fetch anyway
+  // But for example, if you updated a quote at 13:00:10, closed the tab, and opened it at 13:00:20
+  // it will skip the initial quote update, because it hasn't been 30 seconds since the last update (13:00:20 - 13:00:10 = 10)
+  // But since we have refreshInterval = 30/2, it will try to refresh again at 13:00:35 and at 13:00:50
+  // At 13:00:50 it will be 30 seconds since the last update (13:00:50 - 13:00:10 = 40)
+  // having this, we won't have too big time gap between quote updates
+  refreshInterval: PENDING_ORDERS_PRICE_CHECK_POLL_INTERVAL / 2,
+  refreshWhenHidden: false,
+  refreshWhenOffline: false,
+  revalidateOnFocus: false,
+  revalidateIfStale: false,
+  isPaused() {
+    return !document.hasFocus()
+  },
+}
+
+interface UnfillableOrderUpdaterProps {
+  chainId: SupportedChainId
+  order: OrderToCheckFillability
+}
+
+export function UnfillableOrderUpdater({ chainId, order }: UnfillableOrderUpdaterProps): null {
+  const [orderLastTimePriceUpdate, setOrderLastTimePriceUpdate] = useAtom(orderLastTimePriceUpdateAtom)
+  const updatePendingOrderPrices = useSetAtom(updatePendingOrderPricesAtom)
+  const updateIsUnfillableFlag = useUpdateIsUnfillableFlag()
+  const isProviderNetworkUnsupported = useIsProviderNetworkUnsupported()
+  const isWindowVisible = useIsWindowVisible()
+
+  const lastTimePriceUpdate = orderLastTimePriceUpdate[order.id]
+  const lastTimePriceUpdateRef = useRef(lastTimePriceUpdate)
+  lastTimePriceUpdateRef.current = lastTimePriceUpdate
```

Comment:
Is this used anywhere?

Reply:
Wow! Even Rabbit didn't notice that!
Thanks! Fixed